src/celeritas/em/interactor/MuBremsstrahlungInteractor.hh

```diff
@@ -98,16 +121,23 @@ CELER_FUNCTION MuBremsstrahlungInteractor::MuBremsstrahlungInteractor(
     , allocate_(allocate)
     , element_(material.make_element_view(elcomp_id))
     , particle_(particle)
+    , xmin_(std::log(value_as<Energy>(cutoffs.energy(shared.gamma))
+                     / value_as<Energy>(min_cutoff_energy())))
+    , xmax_(std::log(value_as<Energy>(particle_.energy())
+                     / value_as<Energy>(cutoffs.energy(shared.gamma))))
+    , envelope_(value_as<Energy>(cutoffs.energy(shared.gamma))
+                * this->calc_differential_xs(
+                    value_as<Energy>(cutoffs.energy(shared.gamma))))
 {
-    CELER_EXPECT(particle_.energy() >= shared_.min_incident_energy()
-                 && particle_.energy() <= shared_.max_incident_energy());
-    CELER_EXPECT(particle.particle_id() == shared_.ids.mu_minus
-                 || particle.particle_id() == shared_.ids.mu_plus);
+    CELER_EXPECT(particle.particle_id() == shared_.mu_minus
+                 || particle.particle_id() == shared_.mu_plus);
+    CELER_EXPECT(cutoffs.energy(shared.gamma) >= min_cutoff_energy());
+    CELER_EXPECT(particle_.energy() > cutoffs.energy(shared.gamma));
```

Comment:
These conditions are reasonable and practically should work.  Nevertheless, these may be too strict as 1) the `cutoffs` is material-dependent (and also a user may set it to very low which can be lower than the magic threshold 0.9keV), 2) the valid energy range of this model (down to zero) is independent to the gamma cutoffs - do we have a logical protection that this interactor will not be chosen as a discrete process during the process competition or is excluded by the continuous energy loss action (by the stopping range)? - just wonder whether it is fully guaranteed.

Reply:
I _think_ the cutoff value will be set to the max of the cutoff and this magic threshold (see [here](https://github.com/Geant4/geant4/blob/e58e650b32b961c8093f3dd6a2c3bc917b2552be/source/processes/electromagnetic/utils/src/G4EmModelManager.cc#L521) and [here](https://github.com/Geant4/geant4/blob/e58e650b32b961c8093f3dd6a2c3bc917b2552be/source/processes/electromagnetic/muons/src/G4MuBremsstrahlungModel.cc#L109), but I've only skimmed the code). If not, this check should catch it (and we should modify the interactor to not sample an interaction below the threshold). Because the process cross section is zero below the cutoff, we should be guaranteed to never sample this interaction if the incident energy is less than the cutoff.